modules/lang-painless/src/main/java/org/elasticsearch/painless/ir/MemberFieldLoadNode.java

```diff
@@ -51,12 +50,8 @@ public boolean isStatic() {
 
     /* ---- end node data ---- */
 
-    public MemberFieldNode() {
-        // do nothing
-    }
-
     @Override
-    public void write(ClassWriter classWriter, MethodWriter methodWriter, Globals globals, ScopeTable scopeTable) {
+    public void write(ClassWriter classWriter, MethodWriter methodWriter, ScopeTable scopeTable) {
```

Comment:
There's now `MemberFieldLoadNode` and `MemberFieldStoreNode` where `Store` has the following new code:
```java
        if (isStatic == false) {
            methodWriter.loadThis();
        }

        getChildNode().write(classWriter, methodWriter, scopeTable);
```

Why the new code?

Reply:
The new code is to handle putting the value on the stack that will be written to the field. So store requires an expression to generate a value to store where as load does not.